src/components/SearchBar/AbstractSearchBar.js

```diff
@@ -0,0 +1,149 @@
+import  React from 'react';
+import "react-filter-box/lib/react-filter-box.css";
+import "./index.css";
+
+import ReactFilterBox, { SimpleResultProcessing, GridDataAutoCompleteHandler} from "react-filter-box";
+
+ import CheckCircle  from 'react-icons/lib/fa/check-circle';
+ import TimeCircle  from 'react-icons/lib/fa/times-circle';
+ import SearchIcon  from 'react-icons/lib/fa/search';
+ import columnAccessor from "../../utils/columnAccessor";
+ 
+
+//extend this class to add your custom operator
+export class CustomAutoComplete extends GridDataAutoCompleteHandler {
+    
+    // override this method to add new your operator
+    needOperators(parsedCategory) {
+        var result = super.needOperators(parsedCategory);
+        return result.concat(["startsWith"]);
+    }
+
+    //override to custom to indicate you want to show your custom date time
+    needValues(parsedCategory, parsedOperator) {
+        // if(parsedOperator === "after") {
+        //     return [{ customType:"date"}]
+        // }
+
+        return super.needValues(parsedCategory, parsedOperator);
+    }
+}
+
+export class CustomResultProcessing extends SimpleResultProcessing {
+
+    constructor(props, columns = false) {
+        super(props);
+        this.columns = columns;
+    }
+
+    // override this method
+    filter(row, fieldOrLabel, operator, value) {
+
+        const field = this.tryToGetFieldCategory(fieldOrLabel);
+        const searchedValue = value.toLowerCase();
+        let originalValue = (row[field]) ? row[field].toString().toLowerCase() : "";
+
+        if(this.columns) {
+
+            let column = this.columns.find(function(d) {
+                return d.column && d.column === field;
+            });
+
+            if(column) {
+                const formattedValue = columnAccessor(column);
+                originalValue = (formattedValue(row, true)) ? formattedValue(row, true).toString().toLowerCase() : "";  
+            }
+        }
+
+
+        switch(operator) {
+            case "==": return originalValue === searchedValue;
+            case "!=": return originalValue !== searchedValue;
+            case "contains": return originalValue.indexOf(searchedValue) >=0;
+            case "!contains": return originalValue.indexOf(searchedValue) <0;
+            case "startsWith": return  originalValue.startsWith(searchedValue) ;
+            default:
+              return false;
+            
+        }
+    }
+}
+    
+export default class AbstractSearchBar extends React.Component {
```

Comment:
do we need an abstractsearchbar? I think we should have a searchbar and we pass as props handleSearch